src/lib/readmeAPIFetch.ts

```diff
@@ -208,6 +209,102 @@ export async function readmeAPIv1Fetch(
     });
 }
 
+/**
+ * Wrapper for the `fetch` API so we can add rdme-specific headers to all ReadMe API v2 requests.
+ *
+ * @param pathname the pathname to make the request to. Must have a leading slash.
+ * @param fileOpts optional object containing information about the file being sent.
+ * We use this to construct a full source URL for the file.
+ */
+export async function readmeAPIv2Fetch<T extends Command>(
+  this: T,
+  pathname: string,
+  options: RequestInit = { headers: new Headers() },
+  fileOpts: FilePathDetails = { filePath: '', fileType: false },
+) {
+  let source = 'cli';
+  let headers = options.headers as Headers;
+
+  if (!(options.headers instanceof Headers)) {
+    headers = new Headers(options.headers);
+  }
+
+  headers.set(
+    'User-Agent',
+    this.config.userAgent.replace(this.config.name, `${this.config.name}${isGHA() ? '-github' : ''}`),
+  );
+
+  if (!headers.get('accept')) {
+    headers.set('accept', 'application/json');
+  }
+
+  if (isGHA()) {
+    source = 'cli-gh';
+    if (process.env.GITHUB_REPOSITORY) headers.set('x-github-repository', process.env.GITHUB_REPOSITORY);
+    if (process.env.GITHUB_RUN_ATTEMPT) headers.set('x-github-run-attempt', process.env.GITHUB_RUN_ATTEMPT);
+    if (process.env.GITHUB_RUN_ID) headers.set('x-github-run-id', process.env.GITHUB_RUN_ID);
+    if (process.env.GITHUB_RUN_NUMBER) headers.set('x-github-run-number', process.env.GITHUB_RUN_NUMBER);
+    if (process.env.GITHUB_SHA) headers.set('x-github-sha', process.env.GITHUB_SHA);
+
+    const filePath = await normalizeFilePath(fileOpts);
+
+    if (filePath) {
+      /**
+       * Constructs a full URL to the file using GitHub Actions runner variables
+       * @see {@link https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables}
+       * @example https://github.com/readmeio/rdme/blob/cb4129d5c7b51ff3b50f933a9c7d0c3d0d33d62c/documentation/rdme.md
+       */
+      try {
+        const sourceUrl = new URL(
+          `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/blob/${process.env.GITHUB_SHA}/${filePath}`,
+        ).href;
+        headers.set('x-readme-source-url', sourceUrl);
+      } catch (e) {
+        this.debug(`error constructing github source url: ${e.message}`);
+      }
+    }
+  }
+
+  if (isCI()) {
+    headers.set('x-rdme-ci', ciName());
+  }
+
+  headers.set('x-readme-source', source);
+
+  if (fileOpts.filePath && fileOpts.fileType === 'url') {
```

Comment:
ah yikes yeah this would overwrite that. we don't do anything with this header at the moment but i'll look into why i wrote it this way, this is just holdover from the API v1 code